### DISCUSSION TITLE
GIF Not Supported

### DISCUSSION THREAD

**Discussion by ilovepixi** [pixijs/pixijs#7971] [Q&A] [answered]
### Expected Behavior

It is possible to use gif ?

### Possible Solution

`PIXI.Sprite.from('image.gif');`

### Current Behavior

No reaction

**Comment by ivanpopelyshev** [comment]
Here's demo: https://pixiplayground.com/#/edit/wjAO30L8ch8q4RH17uRuS but it doesnt work because `gifuct-js` was changed , author removed UMD version of package. You have to include it through `npm i gifuct-js`, and there you'll get GIF class

There's no other way of doing this.

**Comment by ilovepixi** [comment]
T

> 
> 
> Here's demo: https://pixiplayground.com/#/edit/wjAO30L8ch8q4RH17uRuS but it doesnt work because `gifuct-js` was changed , author removed UMD version of package. You have to include it through `npm i gifuct-js`, and there you'll get GIF class
> 
> There's no other way of doing this.

I can use gif images with other packages, but this is strange why pixijs can't support this.
Hope soon this feature can be add.

**Reply by eXponenta** [reply]
It never will added. GIF is legacy and not supported for direct load to GPU, because require slicing and right time control. Same as use Animation sprite.
Including GIF decoder in runtime for so no-often image format is strange idea.
Core team will not support this.

**Reply by photonstorm** [reply]
I completely understand why you won't support animated gifs. It's a terrible format for GPUs.

But I'm surprised you can't load a gif and have the first frame rendered to a texture? I mean you don't have a JPEG, WEBP or PNG parser built into Pixi, right? So surely you're using the browser's native ability to load those formats, and browsers support GIFs, so what am I missing as to why it doesn't work in a _non-animated_ way?

**Reply by eXponenta** [reply]
@photonstorm 

GIF without animation is not needed.
If we will support non-animated GIF, other will crying why we not support animation.

Easily way to drop support GIF, instead of always explain why we not support animation.

(and we not drop it, we only prevent parse gif)

Pixi is not Phaser and team wan not down to "support anything for anybody'

@ilovepixi 
If you want - open bounty, but anyway officially this is not prefered.

If you are afraid about image weight,  you can easily create a compressed indexed png and results will be simmilar.

**Comment by bigtimebuddy** [comment]
Just an idea to throw out there: someone could build a build-time plugin (maybe for Webpack plugin? or CLI?) to convert a GIF into frames and load as a texture atlas with animations. There are a few tools to extract GIF frames (e.g. gif-extract-frames). I think that might be more practical than runtime solution.

Animated GIF supports pops up occasionally, I don't think it's a core feature because of the low quality of GIF. But if someone wants to come up with a plugin solution, I'm sure others would benefit.

**Reply by eXponenta** [reply]
Problems that this is external tooling, not all use a webpack or other bundlers.
Beginners use UMD packages and i never had a GIF request from skilled devs.

This is why GIF should be completed (support animation)  in main packages, but it requires a third-party plugins that can be problem for pixi like loader or is-mobile, and will increase bundle size.

One benefits of GIF that a it looks like how special designed for pixel art, and more pixel art tools export to it by default.

**Comment by bigtimebuddy** [comment, marked as answer]
I created a PixiJS plugin for rendering GIF animations. Feedback welcome.

### 🍎   Basics

* npm package name `@pixi/gif`
* [Project Repo](https://github.com/pixijs/gif)
* [API Documentation](https://pixijs.io/gif/docs/)

### 📺   Demos

* [Basic](https://pixijs.io/gif/demo/)
* [Multiple Playback](https://jsfiddle.net/bigtimebuddy/gudsxaq2/)

### ⭐   Features

* Supports PixiJS 5+
* Supports canvas rendering
* Control playback including play, stop and seek
* Plugin size is < 13KB minified

**Reply by ilovepixi** [reply]
> 
> 
> I created a PixiJS plugin for rendering GIF animations. Feedback welcome.
> ### 🍎 Basics
> 
>     * npm package name `@pixi/gif`
> 
>     * [Project Repo](https://github.com/pixijs/gif)
> 
>     * [API Documentation](https://pixijs.io/gif/docs/)
> 
> 
> ### 📺 Demos
> 
>     * [Basic](https://pixijs.io/gif/demo/)
> 
>     * [Multiple Playback](https://jsfiddle.net/bigtimebuddy/gudsxaq2/)
> 
> 
> ### ⭐ Features
> 
>     * Supports PixiJS 5+
> 
>     * Supports canvas rendering
> 
>     * Control playback including play, stop and seek
> 
>     * Plugin size is < 13KB minified

@bigtimebuddy This  is awesome, You are the best person i see in my life, i never forget you. you changed my life with pixijs, i wish best for you. thanks.

**Comment by AgustinLaValla** [comment]
Hi @bigtimebuddy . I have this code, similar to the one shown in the documentation:  [https://codesandbox.io/s/frosty-haze-ij165?file=/src/index.js](https://codesandbox.io/s/frosty-haze-ij165?file=/src/index.js)

But I have this error:  
![image](https://user-images.githubusercontent.com/53668963/152694740-b7353bff-d6d4-48b1-a709-8e9c4fdea990.png)

I cannot figure out what's going on. Would you what's wrong? Thanks

**Reply by bigtimebuddy** [reply]
Would you mind posting an issue here: github.com/pixijs/gif

**Reply by AgustinLaValla** [reply]
Sure @bigtimebuddy . Thanks
